src/Units/Unit.java

```diff
@@ -46,27 +47,35 @@ public Unit(Commander co, UnitModel um)
 		if (model.holdingCapacity > 0)
 			heldUnits = new Vector<Unit>(model.holdingCapacity);
 	}
-	
-	public void initTurn()
-	{
+
+	public void initTurn(Location locus) {
 		isTurnOver = false;
 		fuel -= model.idleFuelBurn;
-		if (captureTarget != null && captureTarget.getResident() != this)
-		{
+		if (captureTarget != null && captureTarget.getResident() != this) {
 			captureTarget = null;
 			captureProgress = 0;
 		}
+		if (HP < 9.99) {
+			if (model.canRepairOn(locus) && locus.getOwner() == CO) {
+				int cost = model.moneyCost * Math.min(10 - getHP(), 2)/10; // 1/10th the cost to repair fully
+				if (CO.money >= cost) {
+					CO.money -= cost;
+					alterHP(2);
+				} else if (CO.money >= model.moneyCost/10) {
```

Comment:
It looks a little odd to calculate the cost of healing, and then throw it out the door and recalculate from the model again if the CO can only heal 1 HP (especially since the formula is much simpler this time). I'd be tempted to either use the formula both places (model.moneyCost/10 for 1 and model.moneyCost/5 for 2), or calculate both up front and use the one we can afford.